osgi.specs/docbook/160/service.jakartaws.xml

```diff
@@ -194,8 +216,23 @@ public class SOAPLoggingHandler implements SOAPHandler&lt;SOAPMessageContext&gt;
     
 }</programlisting>
 <para>
- Handlers will be added to the Endpoint in service ranking order, that means that Handlers with a higher service ranking will be called before those with a lower service ranking.
- If two handlers have the same ranking they are sorted according to their service id, having handlers registered later being called later in the handler chain. 
+ Handlers are assembled into a handler chain in service ranking order. This means that Handlers with a higher service ranking will be positioned before those with a lower service ranking in the chain.
+ If two handlers have the same ranking they are sorted according to their service id, with handlers registered later being positioned later in the handler chain. 
+</para>
+<para>
+ This handler chain is then passed to the Jakarta™ XML Web Services implementation. It is important to understand that the Jakarta™ XML Web Services runtime processes the handler chain differently depending on the message direction,
+ as specified in <xref linkend="service.jakartaws-handler-execution"/>:
+ <itemizedlist>
+   <listitem>
+     <para><emphasis>For outbound messages</emphasis>: Handler processing starts with the first handler in the chain and proceeds in the same order as the handler chain. This means the handler with the highest service ranking is processed first.</para>
+   </listitem>
+   <listitem>
+     <para><emphasis>For inbound messages</emphasis>: The order of processing is reversed. Processing starts with the last handler in the chain and proceeds in the reverse order of the handler chain. This means the handler with the lowest service ranking is processed first.</para>
```

Comment:
I would put inbound first as it better matches the request/response flow, and say something like:

```
Inbound messages represent incoming Web Service requests, for example incoming HTTP requests from the internet. For inbound Messages the handlers are processed in ascending order, with lower ranked services executing first.
```

Then for outbound

```
Outbound messages represent outgoing Web Service responses, for example the result of a method call. For outbound messages the handlers are processed in descending order, with higher ranked services executing first.
```

Note that this is the exact reverse behaviour of the Servlet and Jakarta REST whiteboards, where incoming requests are processed by filters in descending order. We should discuss whether this is a good or bad thing.